packages/openapi-v3/src/controller-spec.ts

```diff
@@ -271,31 +256,62 @@ function processSchemaExtensions(
   }
 }
 
+function processSchemaExtensionsForRequestBody(
+  spec: ControllerSpec,
+  schema?: SchemaObject | (ReferenceObject & ISpecificationExtension),
+) {
+  debug('  processing extensions in schema: %j', schema);
+  if (!schema) return;
+
+  assignRelatedSchemas(spec, schema.definitions);
+  delete schema.definitions;
+
+  const tsType = schema.options && schema.options[TS_TYPE_KEY];
+  debug('  %s => %o', TS_TYPE_KEY, tsType);
+  if (tsType) {
+
+    if (!schema.options.isVisited) schema = resolveSchema(tsType, schema);
+    if (schema.$ref) generateOpenAPISchema(spec, tsType, schema.options);
+    delete schema.options;
+    return;
+  }
+  // We don't want a Function type in the final spec.
+  delete schema.options;
+  if (schema.type === 'array') {
+    processSchemaExtensionsForRequestBody(spec, schema.items);
+  } else if (schema.type === 'object') {
+    if (schema.properties) {
+      for (const p in schema.properties) {
+        processSchemaExtensionsForRequestBody(spec, schema.properties[p]);
+      }
+    }
+  }
+}
+
 /**
  * Generate json schema for a given x-ts-type
  * @param spec - Controller spec
  * @param tsType - TS Type
  */
-function generateOpenAPISchema(spec: ControllerSpec, tsType: Function) {
+function generateOpenAPISchema(spec: ControllerSpec, tsType: Function, options?: SchemaOptions) {
```

Comment:
I find this change weird. `spec` if of type `ControllerSpec`, which means it contains specification of multiple controller methods. Typically, each controller method accepts a slightly different request body: `create` accepts model data excluding the PK property, `patchById` accepts model data with all properties optional, and so on. I don't see how a single `SchemaOptions` is going to work for that.

Could you please help me to better understand your intent here?

Reply:
As I am reading through the code, IIUC, `generateOpenAPISchema` is adding schema for `tsType` to `spec.components.schemas` - I guess that starts to explain why you are adding `options` argument.

IIUC, you are trying to support `x-ts-type` together with `schemaOptions`. This is not something we support now, and personally I'd prefer to not support it in the future either.

Reply:
> spec if of type ControllerSpec, which means it contains specification of multiple controller methods. Typically, each controller method accepts a slightly different request body: create accepts model data excluding the PK property, patchById accepts model data with all properties optional, and so on. I don't see how a single SchemaOptions is going to work for that.

Yep `generateOpenAPISchema` generates json schema for a given `x-ts-type` and add it to `spec.components.schemas`